nanoFirmwareFlasher.Library/FirmwarePackage.cs

```diff
@@ -342,7 +360,28 @@ internal async Task<ExitCodes> DownloadAndExtractAsync()
                             Console.ForegroundColor = ConsoleColor.White;
                         }
 
+                        //send app insight on successful download
+
+                        string insightConnectionString = _configuration["iConnectionString"];
+                        string optOut = Environment.GetEnvironmentVariable("DOTNET_CLI_TELEMETRY_OPTOUT");
+
                         stepSuccessful = true;
+
+                        if (!string.IsNullOrEmpty(insightConnectionString) && optOut != "1")
+                        {
+                            TelemetryClient telemetryClient = new TelemetryClient(new Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration()
```

Comment:
what happens if the key is not correct? Need to use a try/catch?

Reply:
It is already in a try catch, loop see line 386. The code was tested using a faulty connectionstring, when this fails the rest of the program continues as it should.